itou/users/models.py

```diff
@@ -134,6 +135,25 @@ def get_duplicates_by_pole_emploi_id(self, prefetch_related_lookups=None):
 
         return result
 
+    def job_seekers_ids_in_user_list(self, user, organization=None):
+        job_seekers_created_by_user = self.filter(created_by=user).values_list("id", flat=True)
+
+        job_application_model = apps.get_model("job_applications", "JobApplication")
+        if organization:
+            job_seekers_created_by_orga = JobSeekerProfile.objects.filter(
+                created_by_prescriber_organization=organization,
+            ).values_list("user_id", flat=True)
+            job_seekers_applications = job_application_model.objects.filter(
+                (Q(sender=user) & Q(sender_prescriber_organization__isnull=True))
+                | Q(sender_prescriber_organization=organization),
+            ).values_list("job_seeker_id", flat=True)
+        else:
+            job_seekers_created_by_orga = User.objects.none()
+            job_seekers_applications = job_application_model.objects.filter(sender=user).values_list(
+                "job_seeker_id", flat=True
+            )
+        return list(job_seekers_created_by_user.union(job_seekers_created_by_orga, job_seekers_applications))
```

Comment:
laisse le consommateur décider s'il souhaite une liste d'ids ou directement le queryset

Reply:
Via un paramètre, comme `User.objects.linked_job_seekers(user, orga, ids_only=True)` pour s'épargner les jointures ?

Ou bien dans tous les cas `linked_job_seekers()` renvoie un queryset ? (mais j'imagine qu'il y aurait à nouveau les problèmes de performances dans ce cas-là)

Reply:
Non, je suis toujours pour un:
```suggestion
        return job_seekers_created_by_user.union(job_seekers_created_by_orga, job_seekers_applications)
```
(avec le nom `linked_job_seeker_ids` vu que c'est un queryset renvoyant des id de job_seekers :) )

Et dans la vue `list_job_seekers` on peut faire un `job_seekers_ids = list(User.objects.linked_job_seeker_ids(request.user, request.current_organization))` pour garder le comportement actuel.